spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java

```diff
@@ -1,13 +1,21 @@
 package com.linkedin.avroutil1.spotbugs;
 
 import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper;
+import org.apache.avro.io.BinaryDecoder;
 import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.DecoderFactory;
 
+import java.io.InputStream;
 import java.io.OutputStream;
 
 public class GoodClass {
 
     public void instantiateBinaryEncoder() {
         BinaryEncoder bobTheEncoder = AvroCompatibilityHelper.newBinaryEncoder( (OutputStream) null);
     }
+
+    public void instantiateBinaryDecoder() {
+        BinaryDecoder bobTheDecoder = AvroCompatibilityHelper.newBinaryDecoder( (InputStream) null);
+        BinaryDecoder robertTheDecoder = DecoderFactory.defaultFactory().createBinaryDecoder(new byte[] {1, 2, 3}, null);
```

Comment:
Why is this considered good? Won't it fail in some Avro versions and therefore be non-portable?

Reply:
this particular invocation would be ambiguous and wont compile without the cast

Reply:
this particular invocation actually works across avro 1.4-10.
its deprecated in ~1.6+, but is actually portable.
also i wanted to test for something that isnt a helper invocation

Reply:
Ok, interesting...